Upgrade webpack-preprocessor for babel-loader 9

### Current behavior

After the new release of `babel-loader` `9.0.0` the installation on a project with `webpack-preprocessor` produces a dependency error.

![Screenshot 2022-10-28 at 12 14 16](https://user-images.githubusercontent.com/1938043/198566548-f8b23e2d-99f6-4c7a-9d52-b6436e52f7aa.png)

https://github.com/cypress-io/cypress/blob/develop/npm/webpack-preprocessor/package.json#L40

### Desired behavior

The installation of the newer babel version should not cause package install issues.

### Test code to reproduce

Install `babel-loader` 9 in a project with the latest `@cypress/webpack-preprocessor`

### Cypress Version

10.11.0

### Node version

v19.0.0

### Operating System

mac Ventura 13

### Debug Logs

```shell
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @cypress/webpack-preprocessor@5.15.0
npm ERR! Found: babel-loader@9.0.0
npm ERR! node_modules/babel-loader
npm ERR!   dev babel-loader@"9.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer babel-loader@"^8.0.2" from @cypress/webpack-preprocessor@5.15.0
npm ERR! node_modules/@cypress/webpack-preprocessor
npm ERR!   dev @cypress/webpack-preprocessor@"5.15.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: babel-loader@8.2.5
npm ERR! node_modules/babel-loader
npm ERR!   peer babel-loader@"^8.0.2" from @cypress/webpack-preprocessor@5.15.0
npm ERR!   node_modules/@cypress/webpack-preprocessor
npm ERR!     dev @cypress/webpack-preprocessor@"5.15.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
```
```


### Other

_No response_

--- a/npm/webpack-preprocessor/package.json
+++ b/npm/webpack-preprocessor/package.json
@@ -62,7 +62,7 @@
   "peerDependencies": {
     "@babel/core": "^7.0.1",
     "@babel/preset-env": "^7.0.0",
-    "babel-loader": "^8.0.2",
+    "babel-loader": "^8.0.2 || ^9",
     "webpack": "^4 || ^5"
   },
   "files": [

